### ISSUE TITLE
Missing hints in code and documentation

### ISSUE THREAD

**Issue by OderDochNicht** [nextcloud/documentation#745]
Steps to reproduce

    New install of NextCloud
    Klick on the "Banner" with warnings
    Klick to the list of security warnings
    Read warning concerning HSTS: The "Strict-Transport-Security" HTTP header is not set to at least "15552000" seconds. For enhanced security, it is recommended to enable HSTS as described in the security tips."
    Klick link "security tips"
    Try to find the related topic (It's the last one in the list.)
    Browser search funktion for "HSTS"
    read article
    Copy, that you're out of the game if you don't have access to Apache config files. At least what the next cloud documentation tells you.
    Google for .htaccess and HSTS.

Expected behaviour

For those who like to install NextCloud on other ressources than a root-server it would be nice to know that it wouldn't be necessary to get into the main config files of apache to improve the security. Googling brings a solution for .htaccess:
Header always add Strict-Transport-Security "max-age=15768000; includeSubDomains; preload"
Actual behaviour

A warning and an article documentation which, in parts, is not helpful for a group of users.
Server configuration

Operating system:
some Linux
Web server:
Apache 2
Database:
MySQL/MariaDB 5.5.38
PHP version:
7.2.3
Nextcloud version: (see Nextcloud admin page)
13.0.1
Updated from an older Nextcloud/ownCloud or fresh install:
fresh
Where did you install Nextcloud from:
zip-file from nextcloud.com

Signing status:
Signing status
admin ``` Login as admin user into your Nextcloud and access http://example.com/index.php/settings/integrity/failed paste the results here. ```
n/a **List of activated apps:**
App list
no additional apps - just clean installation ``` If you have access to your command line run e.g.: sudo -u www-data php occ app:list from within your Nextcloud installation folder ```
no root server, sorry

Nextcloud configuration:
Config report

hey, it's just a clean installation. I uploaded the unpacked files via ftp to the rented server. I pointed FireFox to the Address oft he webspace. NextCloud brought up the config-screen for first login, I inserted the credentials and database information. The first I got was the banner with the warnings.

**Comment by c0fe** [comment]
If I can summarize this, this is to put hints in for .htaccess programming?

**Comment by OderDochNicht** [comment]
Hi c0fe,

often I'm learning by doing.
That was the case, when I tried to install NextCloud.
Besides some Typo3 I'm not a website developer or apache admin. During and after installation I got errors and warnings. It took me quite some time to squeeze helpful info from google - especially because I'm not a mother-tongue.

So, I tried to help others on their way on their learning curve.
If that sound ridiculous to you ...

Best Regrads,
Odo


Am 14.05.2018 um 18:14 schrieb c0fe:

**Comment by joshtrichards** [comment]
That's true that HTTP Strict Transport Security can be enabled via the .htaccess file. 

I think as long as it's added after `#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####` it's fine in terms of the NC integrity check. I think if `htaccess.rewritebase` gets enabled afterward it'll get overwritten though.

I don't really like documenting (in an official form) an approach that has that caveat (it feels fragile).

Side note: I'm unclear why the output of `htaccess.rewritebase` doesn't get stashed in a special section that excludes its output from the integrity check, but also simultaneously isolates it from local changes. But that's outside the scope of this Issue.

*tldr:* It's easy enough to document that `Strict-Transport-Security` can be added not only in the vhost conf, but also .htaccess... but I'm also not sold on doing it in the official docs for reasons. 🤔

**Comment by skjnldsv** [comment]
As @joshtrichards noted, documenting the .htaccess approach feels fragile since it can be overwritten when `htaccess.rewritebase` is enabled. The official recommendation stays with the VirtualHost config.
